internal/strdist/strdist.go

```diff
@@ -125,3 +134,37 @@ func globCost(ar, br rune) Cost {
 	}
 	return Cost{SwapAB: 1, DeleteA: 1, InsertB: 1}
 }
+
+// wildcardPrefixMatch compares whether the prefixes of a and b are equal up
+// to the shortest one. The prefix is defined as the longest substring that
+// starts at index 0 and does not contain a wildcard.
+func wildcardPrefixMatch(a, b string) bool {
+	ai := strings.IndexAny(a, "*?")
+	bi := strings.IndexAny(b, "*?")
+	if ai == -1 {
+		ai = len(a)
+	}
+	if bi == -1 {
+		bi = len(b)
+	}
+	mini := min(ai, bi)
+	return a[:mini] == b[:mini]
+}
+
+// wildcardSuffixMatch compares whether the suffixes of a and b are equal up
+// to the shortest one. The suffix is defined as the longest substring that ends
+// at the string length and does not contain a wildcard.
+func wildcardSuffixMatch(a, b string) bool {
+	ai := strings.LastIndexAny(a, "*?")
+	la := 0
+	if ai != -1 {
+		la = len(a) - ai - 1
```

Comment:
These algorithms are exact opposites of one another, so I'm missing why a choice was made not to implement them as such as well.

For example, if we look at the prefix function, the opposite would be:
```
if ai == -1 {
        ai = 0
}
```
Right? Doing something entirely different besides being more work, it seems, is also unnecessary cognitive load.

Reply:
I am probably thinking of the wrong algorithm but I do not think the above will work. We use the indexes for the prefix because they coincide with the prefix length, which is what we really want. Conversely, for the suffixes we want to compare the lengths of the suffixes to get the minimum one and, as far as I know, we cannot do that using indexes without calculating the deltas first because there is no correspondence between index and length. For example:

```
foo*baz
foobar*b
```
We know that the indexes are 3 and 6 respectively. If we take the minimum of the position then I do not see how to get to the fact that we should check only the last character of both strings because the minimum length of the suffix is 1.

Reply:
We must be talking about different things. I'm not suggesting a change in algorithm, but rather just pointing out that the algorithm was laid out differently between the suffix and prefix version. The actual comparison is exactly the same at the end.

Reply:
I am talking about this sentence for example:
> At first I thought this was wrong, again because I had the prefix logic in mind and didn't realize here it's the minimum of the delta, rather than the minimum of the position

I cannot make it like the prefix logic that uses the position because for suffix position != length of the suffix, I have to use the deltas, unless I am missing something.

Reply:
Yes, I think you are missing something.

What's the difference between `a[len(a)-minl:]` and `a[ai:]`? :)